pkg/server/local_observer.go

```diff
@@ -259,7 +267,7 @@ func (s *LocalObserverServer) GetFlows(
 
 nextFlow:
 	for ; ; i++ {
-		flow, err := flowsReader.Next(server.Context())
+		flow, err := flowsReader.Next(ctx)
```

Comment:
While not introduced by this PR, I just notice it now: Passing down `server.Context()` to the flow reader is wrong. I believe this is another reason we want introduce a per-request context (as stated above).

To demonstrate: The flow reader spawns goroutines to implement `follow` requests. These goroutines must be terminated once the request is finished, which is achieved via context cancellation. If we pass down `server.Context()` here, these per-request goroutines will be kept alive until the server is terminated, which is essentially a goroutine leak.  The context passed down to `flowsReader.Next` must only live as long as the request.